### PR TITLE
Fix util/validate to match Ready without matching NotReady

### PR DESCRIPTION
**Pull request by jgmize**

### CODE REVIEW

File: util/validate [hunk 1]
```diff
@@ -10,7 +10,7 @@ num_nodes="$(( $(cat ./.config.json | jq -r '.phase1.num_nodes') + 1 ))"
 
 while true; do
   total_time=$(( ${total_time} + ${wait_duration} ))
-  hcount=$(kubectl get nodes 2>/dev/null | grep 'Ready' | wc -l) || true
+  hcount=$(kubectl get nodes 2>/dev/null | grep ' Ready' | wc -l) || true
```
Comment (shashidharatd) [inline util/validate]: Can we use grep -w 'Ready' instead?

Fix is good otherwise.
/lgtm

Comment (jgmize) [inline util/validate]: Sure, updated.

**Comment by shashidharatd** [comment]
@kubernetes/kubernetes-anywhere-maintainers, PTAL and merge

**Comment by shashidharatd** [comment]
This PR will eliminate the need for additional checks after deploy (like in [here](https://github.com/shashidharatd/test-infra/blob/master/kubetest/anywhere.go#L146). Also it would be logically correct to start phase3 addons deployment after Nodes are ready.

@pipejakob, Can you PTAL merge this pr please.

**Comment by shashidharatd** [comment]
I just noticed that this PR will break the cluster deployment and whatever the current code is a blessing in disguise. 
`validate` action is called after `deploy-cluster` and before `addons` action. When we deploy a cluster all nodes will be in `NotReady` conditions forever, unless we install CNI plugin like weave=net.

We need to rework on the validation part to be more robust.

**Comment by pipejakob** [comment]
@shashidharatd I completely agree, this change would break clusters using kubeadm for phase2.

@jgmize What was the driving force behind wanting this change? Is there another way we can achieve what you're hoping to do, or were you just trying to fix something that looked incorrect?

**Comment by luxas** [comment]
Why is validate run before addons..?

**Comment by jgmize** [comment]
This PR was filed in response to an offhand comment @luxas during an e2e knowledge sharing session, and #445 appears to be a better fix. Closing in favor of that PR.

**Pull request closed without merging**